src/cluster.c

```diff
@@ -5147,6 +5147,16 @@ int verifyClusterConfigWithData(void) {
     return C_OK;
 }
 
+/* Remove all the shard channel related information not owned by the current shard. */
+static inline void removeAllNotOwnedShardChannelSubscriptions(void) {
+    clusterNode *currmaster = nodeIsMaster(myself) ? myself : myself->slaveof;
```

Comment:
```suggestion
static inline void removeAllNotOwnedShardChannelSubscriptions(void) {
    if (!dictSize(server.pubsubshard_channels)) return;
    clusterNode *currmaster = nodeIsMaster(myself) ? myself : myself->slaveof;
```

I still think this fast path is probably worth it, since we still expect most clusters to not use sharded pubsub.